### PR TITLE
[ExitSpec]: Setup K8S PAI Job ExitSpec

### PR DESCRIPTION
**Pull request by yqwang-ms**
A part of https://github.com/microsoft/pai/issues/3428

### CODE REVIEW

File: src/k8s-frameworkcontroller/deploy/k8s-frameworkcontroller.yaml.template [hunk 1]
```diff
@@ -32,7 +32,18 @@ spec:
     spec:
       containers:
       - name: k8s-frameworkcontroller
-        image: frameworkcontroller/frameworkcontroller:v0.3.0
+        image: wang17/frameworkcontroller:v0.4.1
```
Comment (ydye) [inline src/k8s-frameworkcontroller/deploy/k8s-frameworkcontroller.yaml.template]: Please refer to other service

Comment (abuccts) [inline src/k8s-frameworkcontroller/deploy/k8s-frameworkcontroller.yaml.template]: maybe you also need to change [frameworkbarrier](https://github.com/microsoft/pai/blob/master/src/kube-runtime/build/kube-runtime.dockerfile#L28)

Comment (yqwang-ms) [reply src/k8s-frameworkcontroller/deploy/k8s-frameworkcontroller.yaml.template]: Rollback to orignal image until we have a new released FC.

Comment (yqwang-ms) [reply src/k8s-frameworkcontroller/deploy/k8s-frameworkcontroller.yaml.template]: Rollback to orignal image until we have a new released FC.

**Pull request merged**